src/main/groovy/com/avast/gradle/dockercompose/ComposeExtension.groovy

```diff
@@ -1,144 +1,27 @@
-
-    String dockerExecutable = 'docker'
-
-    String dockerComposeWorkingDirectory = null
-    Duration dockerComposeStopTimeout = Duration.ofSeconds(10)
-
-    ComposeExtension(Project project, ComposeUp upTask, ComposeDown downTask) {
-        this.project = project
-        this.downTask = downTask
-        this.upTask = upTask
-        this.dockerExecutor = new DockerExecutor(this)
-        this.composeExecutor = new ComposeExecutor(this)
-
-        if (OperatingSystem.current().isMacOsX()) {
-            // Default installation is inaccessible from path, so set sensible
-            // defaults for this platform.
-            this.executable = '/usr/local/bin/docker-compose'
-            this.dockerExecutable = '/usr/local/bin/docker'
-        }
-    }
-
-    void isRequiredBy(Task task) {
-        task.dependsOn upTask
-        task.finalizedBy downTask
-        def ut = upTask // to access private field from closure
-        task.getTaskDependencies().getDependencies(task)
-                .findAll { Task.class.isAssignableFrom(it.class) && ((Task) it).name.toLowerCase().contains('classes') }
-                .each { ut.shouldRunAfter it }
-        if (task instanceof ProcessForkOptions) exposeAsEnvironment(task as ProcessForkOptions)
-        if (task instanceof JavaForkOptions) exposeAsSystemProperties(task as JavaForkOptions)
-    }
-
-    Map<String, ServiceInfo> getServicesInfos() {
-        upTask.servicesInfos
-    }
-
-    void exposeAsEnvironment(ProcessForkOptions task) {
-        servicesInfos.values().each { serviceInfo ->
-            serviceInfo.containerInfos.each { instanceName, si ->
-                if (instanceName.endsWith('_1')) {
-                    task.environment << createEnvironmentVariables(serviceInfo.name.toUpperCase(), si)
-                }
-                task.environment << createEnvironmentVariables(instanceName.toUpperCase(), si)
-            }
-        }
-    }
-
-    void exposeAsSystemProperties(JavaForkOptions task) {
-        servicesInfos.values().each { serviceInfo ->
-            serviceInfo.containerInfos.each { instanceName, si ->
-                if(instanceName.endsWith('_1')) {
-                    task.systemProperties << createSystemProperties(serviceInfo.name, si)
-                }
-                task.systemProperties << createSystemProperties(instanceName, si)
-            }
+import org.gradle.util.ConfigureUtil
+
+class ComposeExtension extends ComposeSettings {
+    ComposeExtension(Project project) {
+        super(project, '')
+    }
+
+    private HashMap<String, ComposeSettings> settings = [:]
+
+    def methodMissing(String name, def args) {
+        if (name.startsWith('isRequiredBy') && args[0] instanceof String) {
+            def taskName = name.substring('isRequiredBy'.length())
+            taskName = taskName[0].toLowerCase() + taskName.substring(1)
+            ComposeSettings s = settings.computeIfAbsent(name, { createNested(taskName) })
```

Comment:
I have a hard time figuring out what this exactly does.... is the settings hashmap ever read? I guess all this exists because of the special isRequiredByMyTask syntax ... maybe some comments would help